Contest_API.md

```diff
@@ -1613,7 +1613,7 @@ Properties of judgement objects:
 | id                   | ID        | Identifier of the judgement.
 | submission\_id       | ID        | Identifier of the [submission](#submissions) judged.
 | judgement\_type\_id  | ID ?      | The [verdict](#judgement-types) of this judgement. Required iff judgement has completed.
-| score                | number    | Score for this judgement. Required iff contest:scoreboard_type is `score`.
+| score                | number ?  | Score for this judgement. Required iff contest:scoreboard_type is `score`.
```

Comment:
I don't think this should be nullable. If you don't want to provide a score you must say so in `/access`.

Reply:
Wait but that means that you can not really change the scoring type of a contest, right? If it is `score` you want to give a score (clearly) but if you then change it, you need to update `/access`. Which is 'allowed' but not really preferred.

Reply:
Yes, but changing `scoring_type` is also "allowed" and *definitely not referred. So, that sounds about right to me.

Reply:
OK fair.

But I don't really like that the access depends on the contest type, but that might just be me.

Does this for example also mean that `team.group_ids` should be required (it has a `?` now) since you can do that with access? And some fields on the scoreboard related to pass-fail/score types.

Reply:
> But I don't really like that the access depends on the contest type, but that might just be me.

Are you thinking that `/access` could be a constant for some/most/a system and then irrelevant properties would just be nulled? That would certainly be easier for the system, but it would also make `/access` significantly less useful.

(For the same reason, I'm not so sire `formal_name` actually should be nullable either. Isn't it better to drop it from `/access` if you don't want to provide it? That said, the nullable-or-not question is a lot less important for the contest endpoint since it would have to already be present for `/access` to exist anyway.)

Maybe we need a system level (as opposed to contest level) access endpoint as well. That will give you a superset of what could be offered at a contest on the system. I.e. `/access` would tell you the maximum set of "stuff" that a give API provider could ever offer and `/contests/<id>/access` would tell you what any particular contest is *actually* going to offer. I'm only 50% joking.

Reply:
Joking aside we should also change group_ids and the scoreboard properties to not be billable if we go this way

Reply:
Yes, agreed.

Reply:
If a property isn't in /access then you're guaranteed not to see one, but I was assuming that we wouldn't guarantee the reverse. e.g. If org.logo is not returned from /access then you are assured that the provider doesn't support logos or there are none in this contest. If org.logo is returned then the provider definitely supports them - but may not have scanned every org in the current contest to see if there are any. Or maybe there aren't any currently but it doesn't want to commit that there won't be any later, or hasn't done anything to block an inadvertent change.

Said another way, it gets more and more complex to handle account-specific and then contest-specific /access, and I personally don't expect all of our current API providers to get this far and get it right. In these cases there isn't even any benefit from implementing it - e.g. clients should check if this is a scoring contest by looking at contest.scoreboard_type not whether judgements.score exists, and in all of the rest the client would need to have null checks anyway.

Reply:
> If a property isn't in /access then you're guaranteed not to see one, but I was assuming that we wouldn't guarantee the reverse

Right, but only because if some of the properties listed in `/access` are nullable then they could be null (duh) and since we say that null and missing must be equal, they must be able to be missing.

Reply:
I think this one should also not be nullable: if you say in `/access` that you'll provide it, then you must always.